brigade-worker/src/k8s.ts

```diff
@@ -325,15 +325,14 @@ export class JobRunner implements jobs.JobRunner {
 
     // Add secret volume
     this.runner.spec.volumes = [
-      { name: secName, secret: { secretName: secName } } as kubernetes.V1Volume,
-      { name: "vcs-sidecar", emptyDir: {} } as kubernetes.V1Volume
+      { name: secName, secret: { secretName: secName } } as kubernetes.V1Volume
     ];
     this.runner.spec.containers[0].volumeMounts = [
-      { name: secName, mountPath: "/hook" } as kubernetes.V1VolumeMount,
-      { name: "vcs-sidecar", mountPath: mountPath } as kubernetes.V1VolumeMount
+      { name: secName, mountPath: "/hook" } as kubernetes.V1VolumeMount
     ];
 
-    if (job.useSource && project.repo.cloneURL) {
+    this.runner.spec.initContainers = [];
+    if (job.useSource && project.repo.cloneURL && project.kubernetes.vcsSidecar) {
```

Comment:
nit: this is making me slightly nervous as we're checking a boolean and a string in the same way here. Could we check for value and length? Something along the lines of [this](https://codereview.stackexchange.com/a/5710)?

While this is by no means reason to bump this PR again, I'd love to get some linters and checkers in place to prevent this kinds of things in the future, and would like to fix this in #739 .